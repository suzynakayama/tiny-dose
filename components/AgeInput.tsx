import { View, Text, StyleSheet } from "react-native";
import React from "react";
import SwitchSelector from "react-native-switch-selector";
import { ageOptions } from "@/app/medicineCalculator/medicineCalculator.contants";
import { SwitchType } from "@/app/medicineCalculator/medicineCalculator.types";

// This component allows the user to select the age of the child
// It uses a SwitchSelector to toggle between the ageOptions array

interface AgeInputProps {
	changeSwitch: (value: string, type: SwitchType) => void;
}

export default function AgeInput({ changeSwitch }: AgeInputProps) {
	return (
		<>
			<Text>Child Age:</Text>
			<SwitchSelector
				options={ageOptions}
				initial={0}
				onPress={(value: string) => changeSwitch(value, "age")}
				fontSize={14}
				selectedTextStyle={{ fontWeight: "bold" }}
				buttonColor="#72CFF8"
				textColor="#333"
				backgroundColor="#cde6f0"
				style={styles.ageSelector}
			/>
		</>
	);
}

const styles = StyleSheet.create({
	ageSelector: {
		maxWidth: 250,
		padding: 3,
	},
});
