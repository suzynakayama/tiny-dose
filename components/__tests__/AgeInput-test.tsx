import React from 'react'
import { render, screen } from '@testing-library/react-native'
import AgeInput from '../AgeInput'

test('renders correctly', () => {
  const changeSwitchMock = jest.fn()
  const AgeInputComponent = render(<AgeInput changeSwitch={changeSwitchMock} />)

  const ageText = screen.getByText('Child Age:')
  expect(ageText).toBeOnTheScreen()

  const switchSelector = screen.getByTestId('age-selector')
  expect(switchSelector).toBeOnTheScreen()

  // Check if the switch has the correct options
  const options = ['1-6 months', 'over 6 months']
  options.forEach(option => {
    expect(screen.getByText(option)).toBeOnTheScreen()
  })

  // Snapshot testing
  const ageInputJson = AgeInputComponent.toJSON()
  expect(ageInputJson).toMatchSnapshot()
})
