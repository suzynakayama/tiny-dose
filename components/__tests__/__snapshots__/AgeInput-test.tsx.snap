// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <Text>
    Child Age:
  </Text>,
  <View
    accessibilityLabel="age-selector"
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "maxWidth": 250,
          "padding": 3,
        },
      ]
    }
    testID="age-selector"
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          {
            "backgroundColor": "#f6bdc0",
            "borderRadius": 50,
            "height": 40,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderColor": "#C9C9C9",
              "borderRadius": 50,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityLabel="between-1-and-6-months-old"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 14,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  {
                    "fontWeight": "bold",
                    "textShadowColor": "#031F72",
                    "textShadowOffset": {
                      "height": 1,
                      "width": 1,
                    },
                    "textShadowRadius": 2,
                  },
                ]
              }
            >
              1-6 months
            </Text>
          </View>
          <View
            accessibilityLabel="over-6-months-old"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#333",
                    "fontSize": 14,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  {},
                ]
              }
            >
              over 6 months
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
